core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/dot-uve-toolbar/dot-uve-toolbar.component.ts

```diff
@@ -127,13 +138,65 @@ export class DotUveToolbarComponent {
     readonly $actions = this.#store.workflowLoading;
     readonly $workflowLoding = this.#store.workflowLoading;
 
+    protected readonly date = new Date();
+
+    defaultDevices = DEFAULT_DEVICES;
+
+    handleViewParamsEffect = effect(
+        () => {
+            const devices = this.$devices();
+
+            //We want to be sure that we have devices
+            if (!devices.length) return;
+
+            // Get the device and orientation from the URL
+            const { device: deviceInode, orientation } = this.#store.viewParams();
+
+            // So we dont open the editor in a device
+            if (!this.#store.$isPreviewMode() && (deviceInode || orientation)) {
+                this.#store.patchViewParams({ device: null, orientation: null });
+
+                return;
+
+                // If we are in preview mode and we dont have a device, set the default device
+            } else if (this.#store.$isPreviewMode() && !deviceInode) {
+                this.#store.patchViewParams({ device: 'default', orientation: null });
+
+                return;
+            }
+
+            // Find the device in the devices list
+            const device = this.$devices().find((d) => d.inode === deviceInode);
+
+            // If we have a device, set it in the store
+            if (device) {
+                this.#store.setDevice(device, orientation);
+            } else {
+                // If we dont have a device, clear the device and social media
+                this.#store.clearDeviceAndSocialMedia();
+            }
+        },
+        {
+            allowSignalWrites: true
+        }
+    );
```

Comment:
This should not be an effect; it should happen just once in the ngOnInit (Shell Component). This `effect` checks that the `device` in the `QP` is valid. 

However, we only read the `QP` once, after that the user can only select valid `devices` so we shouldn't do this validation every time a `device` or `viewParams` changes.

Reply:
We should have an inner rule of only using this `allowSignalWrites: true` when it is strictly necessary; otherwise we could be doing something wrong

Reply:
This will listen to changes in the QP, because I need to change the QP to maintain the state, this helps initialize the preview mode using the QP and also share the link if needed.

Reply:
After discussing it we got a simpler solution